### PR TITLE
Reasource Access Handler

### PR DESCRIPTION
**Pull request by TheTechArch**
Added support for resource authorization 
Upgraded to .net 6

### CODE REVIEW

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Constants/AltinnXacmlUrns.cs [hunk 1]
```diff
@@ -39,5 +39,15 @@ public static class AltinnXacmlUrns
         /// Get the value task id
         /// </summary>
         public const string TaskId = "urn:altinn:task";
+
+        /// <summary>
+        /// Get the value resourceId
+        /// </summary>
+        public const string ResourceId = "urn:resourceregistry:resourceid";
```
Comment (jonkjetiloye) [inline src/Altinn.Common.PEP/Altinn.Common.PEP/Constants/AltinnXacmlUrns.cs:46]: So far for changes made in access-management we have used the urn: urn:altinn:resourceregistry
as AttributeId for match for the resourceid. We should decide on a common value here

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Authorization/ResourceAccessHandler.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Altinn.Common.PEP.Authorization
+{
+    /// <summary>
+    /// AuthorizationHandler that is created for handling access to app.
```
Comment (jonkjetiloye) [inline src/Altinn.Common.PEP/Altinn.Common.PEP/Authorization/ResourceAccessHandler.cs:15]: Documentation of class and initializer could use an update (AppAccessHandler ref)

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Constants/AltinnXacmlUrns.cs (hunk 1 above)
Comment (jonkjetiloye) [reply src/Altinn.Common.PEP/Altinn.Common.PEP/Constants/AltinnXacmlUrns.cs:46]: ResourceId is also a bit generic property name. Should this be ResourceRegistryId?

Comment (TheTechArch) [reply src/Altinn.Common.PEP/Altinn.Common.PEP/Constants/AltinnXacmlUrns.cs]: I changed to match your value in access-management

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Authorization/ResourceAccessHandler.cs (hunk 2 above)
Comment (TheTechArch) [reply src/Altinn.Common.PEP/Altinn.Common.PEP/Authorization/ResourceAccessHandler.cs:15]: Fixed

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs [hunk 3]
```diff
@@ -27,6 +27,7 @@ public static class DecisionHelper
         private const string ParamApp = "app";
         private const string ParamOrg = "org";
         private const string ParamAppId = "appId";
+        private const string ParamWho = "who";
```
Comment (jonkjetiloye) [inline src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs:30]: Not a big fan of the "who" parameter. Actor has been mentioned as a 
possibility

**Review by jonkjetiloye — APPROVED** [review]

File: src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs (hunk 3 above)
Comment (TheTechArch) [reply src/Altinn.Common.PEP/Altinn.Common.PEP/Helpers/DecisionHelper.cs]: Fixed

**Pull request merged**